src/pages/registration/resource_type_tab/components/RelatedResultsField.tsx

```diff
@@ -0,0 +1,114 @@
+import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
+import { Box, Button, List, Typography } from '@mui/material';
+import { FieldArray, FieldArrayRenderProps, move, useFormikContext } from 'formik';
+import { Trans, useTranslation } from 'react-i18next';
+import { DegreeRegistration } from '../../../../types/publication_types/degreeRegistration.types';
+import { ResourceFieldNames } from '../../../../types/publicationFieldNames';
+import {
+  ConfirmedDocument,
+  createEmptyUnconfirmedDocument,
+  UnconfirmedDocument,
+} from '../../../../types/registration.types';
+import { dataTestId } from '../../../../utils/dataTestIds';
+import { RelatedResultItem } from '../sub_type_forms/research_data_types/RelatedResultItem';
+import { SearchRelatedResultField } from './SearchRelatedResultField';
+
+export const RelatedResultsField = () => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = useFormikContext<DegreeRegistration>();
+
+  const related = values.entityDescription.reference?.publicationInstance.related ?? [];
+
+  const removeRelatedItem = (indexToRemove: number) => {
+    const newRelated = related?.filter((_, thisIndex) => thisIndex !== indexToRemove);
+    setFieldValue(ResourceFieldNames.PublicationInstanceRelated, newRelated);
+  };
+
+  const moveRelatedResult = (newSequence?: number, oldSequence?: number) => {
+    if (!newSequence || !oldSequence) return;
+
+    const oldIndex = related.findIndex((document) => document.sequence === oldSequence);
+    const minNewIndex = 0;
+    const maxNewIndex = related ? related.length : 0;
+
+    const newIndex =
```

Comment:
Dette virker ganske komplisert og innviklet når problemet bare er å bytte plassering av to elementer. Går man for en trippel ternary burde man ha en god grunn, da det er ekkelt å gjøre endringer/bugfikser her senere. Alle artistiske resultat har lignende sortering, og der er det bare løst med en one-liner med `move()` fra formik  (se feks `ArtisticMusicPerformanceForm.tsx`). I tilfelle her måtte vi kanskje i tillegg satt `sequence` da, men det er en enkel iterasjon etter elementet er riktig plassert.
Ikke sikkert vi bør bruke mer tid på det nå da, men husk til senere å stille spørsmål til hvorfor man gjør det man gjør, og om det kan forenkles :)